core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/Changeset.java

```diff
@@ -127,7 +127,21 @@ public void close() throws SailException {
 		refbacks = null;
 		prepend = null;
 		observed = null;
+		if (approved instanceof AutoCloseable) {
+			try {
+				((AutoCloseable) approved).close();
+			} catch (Exception e) {
+				throw new SailException(e);
+			}
+		}
 		approved = null;
+		if (deprecated instanceof AutoCloseable) {
+			try {
+				((AutoCloseable) deprecated).close();
+			} catch (Exception e) {
+				throw new SailException(e);
+			}
+		}
 		deprecated = null;
```

Comment:
This is interesting. I presume that this is for closing the overflow model. I think the overflow model has some mechanism for automatically closing it once it's unreachable, but if we can explicitly close it then that is much more performant.

I would recommend using try finally. Something like:
```
try{
	if (approved instanceof AutoCloseable) {
			try {
				((AutoCloseable) approved).close();
			} catch (Exception e) {
				throw new SailException(e);
			}
		}
		approved = null;
} finally {
if (deprecated instanceof AutoCloseable) {
			try {
				((AutoCloseable) deprecated).close();
			} catch (Exception e) {
				throw new SailException(e);
			}
		}
		deprecated = null;
}
```

Maybe even introduce two local variables for `approved` and `deprecated` so that these can be set to null first and then closed after everything is set to null.